include/exadg/structure/preconditioners/multigrid_preconditioner.cpp

```diff
@@ -128,25 +149,34 @@ template<int dim, typename Number>
 void
 MultigridPreconditioner<dim, Number>::update_operators()
 {
-  PDEOperatorNonlinear const * pde_operator_nonlinear =
-    dynamic_cast<PDEOperatorNonlinear const *>(pde_operator);
-
-  VectorType const & vector_linearization = pde_operator_nonlinear->get_solution_linearization();
-
-  // convert Number --> MultigridNumber, e.g., double --> float, but only if necessary
-  VectorTypeMG         vector_multigrid_type_copy;
-  VectorTypeMG const * vector_multigrid_type_ptr;
-  if(std::is_same<MultigridNumber, Number>::value)
+  if(data.unsteady)
   {
-    vector_multigrid_type_ptr = reinterpret_cast<VectorTypeMG const *>(&vector_linearization);
+    set_time(pde_operator->get_time());
```

Comment:
Did you figure out the precise reason for `set_time()`? This would be important because if time does not have an impact on the linear system of equations, we should avoid updating the preconditioner (for reasons of computational costs).

Reply:
So fine? This is only triggered if `unsteady` evaluates to true ... and the time is a double ... so probably thats not the problem is it? the time is never used to trigger a preconditioner update? and updating the time variable is not costly, is it?
Anyways I have not found out what the `time` changes. I can think of changing a stiffness coefficient over time (initial damping and initially stiffer springs help a ton in cardiovasc scenarios!). For now, I do not see a reason this is needed -  maybe it is motivated by someone expecting `update` to update the time of the operators as well, which in itself sounds reasonable to me :)

Reply:
Then, I suggest to keep it to be prepared for the future / prevent foreseeable bugs.